src/modules/hip/kernel/tensor.cpp

```diff
@@ -1,249 +1,379 @@
+    output[dst_idx] = input[src_idx];
 }
 
-extern "C" __global__ void tensor_convert_bit_depth_u8s8(   const unsigned int tensorDimension,
-                     unsigned char* input,
-                     char* output,
-                    const unsigned int a,
-                    const unsigned int b,
-                    const unsigned int c
-)
+extern "C" __global__ void tensor_convert_bit_depth_u8s8(const unsigned int tensorDimension,
+                                                         unsigned char *input,
+                                                         char *output,
+                                                         const unsigned int a,
+                                                         const unsigned int b,
+                                                         const unsigned int c)
 {
-    int id_x = hipBlockIdx_x *hipBlockDim_x + hipThreadIdx_x;
-    int id_y = hipBlockIdx_y *hipBlockDim_y + hipThreadIdx_y;
-    int id_z = hipBlockIdx_z *hipBlockDim_z + hipThreadIdx_z;
-    
-    if (id_x >= a || id_y >= b || id_z >= c) return;
-    
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if (id_x >= a || id_y >= b || id_z >= c)
+    {
+        return;
+    }
+
     int pixIdx = id_y * c * a + id_x * c + id_z;
-    
     output[pixIdx] = (char)(input[pixIdx] - 128);
 }
 
-extern "C" __global__ void tensor_convert_bit_depth_u8u16(   const unsigned int tensorDimension,
-                     unsigned char* input,
-                     unsigned short* output,
-                    const unsigned int a,
-                    const unsigned int b,
-                    const unsigned int c
-)
+extern "C" __global__ void tensor_convert_bit_depth_u8u16(const unsigned int tensorDimension,
+                                                          unsigned char *input,
+                                                          unsigned short *output,
+                                                          const unsigned int a,
+                                                          const unsigned int b,
+                                                          const unsigned int c)
 {
-    int id_x = hipBlockIdx_x *hipBlockDim_x + hipThreadIdx_x;
-    int id_y = hipBlockIdx_y *hipBlockDim_y + hipThreadIdx_y;
-    int id_z = hipBlockIdx_z *hipBlockDim_z + hipThreadIdx_z;
-    
-    if (id_x >= a || id_y >= b || id_z >= c) return;
-    
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if (id_x >= a || id_y >= b || id_z >= c)
+    {
+        return;
+    }
+
     int pixIdx = id_y * c * a + id_x * c + id_z;
-    
     output[pixIdx] = (unsigned short)(input[pixIdx] * 257);
```

Comment:
What is the logic of this? Why is it multiplied by 257

Reply:
@rrawther its used to convert a 0-255 range U8 to a 0-65535 range U16 (255 * 257 = 65535)